service/src/main/kotlin/fi/espoo/evaka/pis/controllers/EmployeeController.kt

```diff
@@ -157,6 +159,7 @@ class EmployeeController(private val accessControl: AccessControl) {
                 )
 
                 it.upsertEmployeeDaycareRoles(id, body.daycareIds, body.role)
+                it.upsertEmployeeMessageAccount(id)
```

Comment:
Does this now create personal messaging accounts even when `List<DaycareId>` is empty or when the added role is `STAFF`? Shouldn't messaging accounts be created only for `UNIT_SUPERVISOR` and `SPECIAL_EDUCATION_TEACHER` roles?

Reply:
Added validation for empty list. Roles are apparently as intended.